src/modules/lua_mtev_crypto.c

```diff
@@ -135,7 +138,23 @@ mtev_lua_crypto_x509_index_func(lua_State *L) {
     EVP_PKEY *pkey;
     pkey = X509_get_pubkey(cert);
     if (pkey == NULL) return 0;
-    else if (EVP_PKEY_base_id(pkey) == EVP_PKEY_RSA) {
+#if OPENSSL_VERSION_NUMBER >= _OPENSSL_VERSION_3_0_0
+    const int key_bits = EVP_PKEY_get_bits(pkey);
+    const int key_type = EVP_PKEY_get_base_id(pkey);
+
+    if (key_bits > 0) {
+      switch(key_type) {
+      case EVP_PKEY_RSA:
+      case EVP_PKEY_DSA:
+          lua_pushinteger(L, key_bits);
+          __attribute__((fallthrough));
```

Comment:
```suggestion
```

don't want to use a GNU extension, can upgrade to ``C23`` though, and use ``[[fallthrough]]``

Reply:
Addressed by adding an explicit `break` for each case.